server/discordbot.py

```diff
@@ -0,0 +1,92 @@
+from urllib import parse
+import asyncio
+import discord
+from discord.ext import commands
+from discord.utils import escape_markdown
+from discord.errors import Forbidden, HTTPException
+
+class Bridgebot(commands.Bot):
+    """
+    The AO2 Discord bridge self.
+    """
+    def __init__(self, server, target_chanel, hub_id, area_id):
+        super().__init__(command_prefix='$')
+        self.server = server
+        self.pending_messages = []
+        self.hub_id = hub_id
+        self.area_id = area_id
+        self.target_channel = target_chanel
+
+    @staticmethod
+    def loop_it_forever(loop):
+        loop.run_forever()
+        loop.close()
+
+    async def init(self, token):
+        '''Starts the actual bot'''
+        print('Trying to start the Discord Bridge bot...')
+        try:
+            await self.start(token)
+        except Exception as e:
+            print(e)
+            raise
+    
+    def queue_message(self, name, message, charname):
+        base = None
+        avatar_url = None
+        if "base_url" in self.server.config["bridgebot"]:
+            base = self.server.config["bridgebot"]["base_url"]
+        if base != None:
+            avatar_url = base + parse.quote("characters/" + charname + "/char_icon.png")
+        self.pending_messages.append([name, message, avatar_url])
+
+    async def on_ready(self):
+        print('Discord Bridge Successfully logged in.')
+        print('Username -> ' + self.user.name)
+        print('ID -> ' + str(self.user.id))
+        self.guild = self.guilds[0]
+        self.channel = discord.utils.get(self.guild.text_channels, name=self.target_channel)
+        await self.wait_until_ready()
+
+        while True:
+            if len(self.pending_messages) > 0:
+                await self.send_char_message(*self.pending_messages.pop())
+
+            await asyncio.sleep(max(0.1, self.server.config["bridgebot"]["tickspeed"]))
```

Comment:
you probably want to eliminate polling here - the ideal solution is something like this:

```python3
async for message in self.wait_for_messages():
    await self.send_char_message(*message)
```

Then you make an [`asyncio.Event`](https://docs.python.org/3/library/asyncio-sync.html#asyncio.Event):
```python3
def __init__(self, ...)
    self._pending_msgs_evt = asyncio.Event()
    # ...

async def wait_for_messages(self):
    while True:
        await self._pending_msgs_evt.wait()
        yield self._pending_msgs.pop()
        self._pending_msgs_evt.clear()
```

Finally, to add to the queue:

```python3
def enqueue_message(self, message, ...):
    # ...
    self._pending_msgs.append(message)
    self._pending_msgs_evt.set()
```

There is also a class called [`asyncio.Queue`](https://docs.python.org/3/library/asyncio-queue.html#asyncio.Queue), but it is used for task tracking for parallel jobs and so it is overkill for this case.